src/main.cpp

```diff
@@ -5176,8 +5185,10 @@ bool static ProcessMessage(CNode* pfrom, string strCommand, CDataStream& vRecv,
             if (nEvicted > 0)
                 LogPrint("mempool", "mapOrphan overflow, removed %u tx\n", nEvicted);
         } else {
-            assert(recentRejects);
-            recentRejects->insert(tx.GetHash());
+            if (!state.CorruptionPossible()) {
+                assert(recentRejects);
+                recentRejects->insert(tx.GetHash());
```

Comment:
This is minor, but shouldn't we put a similar guard up above, where failed-orphan transactions get added to `recentRejects`?  Presumably failed orphans should also only be added to `recentRejects` if the corruption field is not set?